src/framework/base/dropdown/BaseDropdown.jsx

```diff
@@ -164,17 +156,44 @@ export default class BaseDropdown extends Component {
     });
   }
 
-  enableScrolling() {
-    if ($(document).height() > $(window).height()) {
-      const scrollTop = parseInt($('html').css('top'), 10);
-      $('html').removeClass('html--noscroll');
-      $('html,body').scrollTop(-scrollTop);
+  chooseOpeningDirection() {
+    const $input = $(this.refs.input);
+
+    if (!$input.offset()) {
+      return;
+    }
+
+    const $optionList = $(this.refs.optionList);
+
+    const roomBelow =
+      (
+        $(window).height() - $input.offset().top - $input.parent().outerHeight()
+      ) + $(window).scrollTop();
+    const roomAbove = $input.offset().top - $(window).scrollTop();
+
+    const EXTRA_SPACE = 10;
```

Comment:
I think this constant makes sense only in the context of `chooseOpeningDirection`. I don't know how to name it if it is in the constructor.